### DISCUSSION TITLE
Negative stack id and pid error when profiling ustack event

### DISCUSSION THREAD

**Discussion by 12ff54e** [bpftrace/bpftrace#2760] [Q&A]
Hi, 

My linux server has an unusual high cpu usage and htop shows nothing. So I first check the kernel stack using `bpftrace -e 'profile:hz:99 /cpu == 0/ { @[kstack] = count(); }'` but most of the time it's empty. When I profile user-level stack it reported an error
```
ERROR: failed to look up stack id -17 (pid -1): -1
```
then prints out lots of stack info with addresses only. What does this error message mean? I try google it but no luck.

**Comment by viktormalik** [comment]
This error appears when bpftrace cannot find an entry in its internal map for keeping stack traces. What's interesting here is the error code, -17 is `-EEXISTS`. The only place where `EEXISTS` is returned from `bpf_map_lookup_elem` is when `BPF_F_REUSE_STACKID` is set but we should never set that flag (at least per [this comment](https://github.com/iovisor/bpftrace/blob/master/src/map.cpp#L96)).

Would be nice if we could reproduce this to investigate more (the above command works well on my localhost).

**Reply by 12ff54e** [reply]
Thanks for your swift reply! I installed bpftrace using apt and the version number is v0.14.0 (maybe a bit old). I will try with the latest verion to check if this error persist on my server.

**Reply by viktormalik** [reply]
Yes, that is certainly the first thing to try.

**Reply by 12ff54e** [reply]
The lasteat version works without that error message. However the user-level stack info is still consist of address only, any suggestion on how do I can get more infomation out of it? This is my 1st time using bpftrace and I'm completely a noob about it.

Thanks
